src/FritzBox/BGimage.php

```diff
@@ -134,13 +134,96 @@ public function getBackgroundImage($quickdials)
             }
             imagettftext($this->bgImage, 20, 0, $posX, $posY, $this->textColor, $this->font, $quickdial);
         }
-        
         ob_start();
         imagejpeg($this->bgImage, null, 100);
-        $content = ob_get_contents();
-        ob_end_clean();
+        $content = ob_get_clean();
 
         return $content;
     }
 
+    /**
+     * Returns a well-formed body string, which is accepted by the FRITZ!Box for uploading
+     * a background image. Guzzle's multipart option does not work on this interface. If
+     * this changes, this function can be replaced.
+     * 
+     * @param string $sID
+     * @param string $phone
+     * @param string $image
+     * @return string
+     */
+    private function getBody($sID, $phone, $image)
+    {
+        $boundary = '--' . sha1(uniqid());
+        $imageSize = strlen($image);
+
+        $body = <<<EOD
+$boundary
+Content-Disposition: form-data; name="sid"
+Content-Length: 16
+
+$sID
+$boundary
+Content-Disposition: form-data; name="PhonebookId"
+Content-Length: 3
+
+255
+$boundary
+Content-Disposition: form-data; name="PhonebookType"
+Content-Length: 1
+
+1
+$boundary
+Content-Disposition: form-data; name="PhonebookEntryId"
+Content-Length: 3
+
+$phone
```

Comment:
Is phone always length 3?

Reply:
yes, internal number must between 610 and 615
see line 200: `$numberRange = range(strval(610), strval(615));`